x-pack/plugins/cases/server/client/user_actions/users.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UserProfile } from '@kbn/security-plugin/common';
+import type { GetCaseUsersResponse } from '../../../common/api';
+import { GetCaseUsersResponseRt } from '../../../common/api';
+import type { OwnerEntity } from '../../authorization';
+import { Operations } from '../../authorization';
+import { createCaseError } from '../../common/error';
+import type { CasesClient } from '../client';
+import type { CasesClientArgs } from '../types';
+import type { GetUsersRequest } from './types';
+import { getUserProfiles } from '../cases/utils';
+
+type User = GetCaseUsersResponse['users'][number];
+
+export const getUsers = async (
+  { caseId }: GetUsersRequest,
+  casesClient: CasesClient,
+  clientArgs: CasesClientArgs
+): Promise<GetCaseUsersResponse> => {
+  const {
+    services: { userActionService },
+    logger,
+    authorization,
+    securityStartPlugin,
+  } = clientArgs;
+
+  try {
+    const users: User[] = [];
+
+    // ensure that we have authorization for reading the case
+    await casesClient.cases.resolve({ id: caseId, includeComments: false });
```

Comment:
Do we need to add the currently assigned users to the `participants`?

Reply:
I thought that the UI should do that. Yes, it is better if we add them as participants.  Users of type `users` will be fewer and the distinction will make more sense.